datafusion/src/physical_plan/functions.rs

```diff
@@ -543,6 +553,28 @@ macro_rules! invoke_if_unicode_expressions_feature_flag {
     };
 }
 
+///Function volatility determines when a function can be inlined or in the future have evaluations elided when the arguments are the same
+///Immutable - a pure function which always remains the same
+///Stable - Maybe not required?. Used for functions which can modify the db but are stable for multiple calls in statement
+///Volatile - Functions where output can vary for each call.
+///For more information see https://www.postgresql.org/docs/current/xfunc-volatility.html
```

Comment:
```suggestion
/// Determines when a function can be inlined or in the future have evaluations elided when the arguments are the same
/// For more information see https://www.postgresql.org/docs/current/xfunc-volatility.html
```

We can use the description of each item to document the enum